homeassistant/components/androidtv/media_player.py

```diff
@@ -397,34 +325,24 @@ def __init__(
         self,
         aftv,
         name,
-        apps,
-        get_sources,
-        turn_on_command,
-        turn_off_command,
-        exclude_unnamed_apps,
-        screencap,
+        dev_type,
+        entry_id,
+        unique_id,
     ):
         """Initialize the Android TV / Fire TV device."""
         self.aftv = aftv
+        self._entry_id = entry_id
         self._attr_name = name
-        self._app_id_to_name = APPS.copy()
-        self._app_id_to_name.update(apps)
-        self._app_name_to_id = {
-            value: key for key, value in self._app_id_to_name.items() if value
-        }
-
-        # Make sure that apps overridden via the `apps` parameter are reflected
-        # in `self._app_name_to_id`
-        for key, value in apps.items():
-            self._app_name_to_id[value] = key
-        self._get_sources = get_sources
-        self._attr_unique_id = self.aftv.device_properties.get("serialno")
-
-        self.turn_on_command = turn_on_command
-        self.turn_off_command = turn_off_command
+        self._attr_unique_id = unique_id
+        self._attr_device_info = self._get_device_info(dev_type, unique_id)
```

Comment:
```suggestion
        info = aftv.device_properties
        model = info.get(ATTR_MODEL)
        self._attr_device_info = {
            "connections": {
                (
                    CONNECTION_NETWORK_MAC,
                    format_mac(info.get(PROP_ETHMAC) or info.get(PROP_WIFIMAC, "")),
                )
            },
            ATTR_IDENTIFIERS: {(DOMAIN, unique_id)},
            ATTR_MANUFACTURER: info.get(ATTR_MANUFACTURER),
            ATTR_MODEL: f"{model} ({dev_type})" if model else dev_type,
            ATTR_NAME: name,
            ATTR_SW_VERSION: info.get(ATTR_SW_VERSION),
        }
```

Reply:
It is correct set the `dict` items also if they are empty (e.g. `ATTR_SW_VERSION: info.get(ATTR_SW_VERSION)`)?

Reply:
If version is available, it will display in the device info. If not, version will not be displayed at all.

Reply:
Also refer to `model` and `connections` (we could have `"connections": {CONNECTION_NETWORK_MAC, ""`)